felix/bpf/conntrack/scanner.go

```diff
@@ -156,14 +167,29 @@ func (s *Scanner) Scan() {
 				"entry": ctVal,
 			}).Debug("Examining conntrack entry")
 		}
+		val = cleanupv1.Value{}
```

Comment:
Better to use if/else instead of unconditionally setting the variable and then overwriting it.

Reply:
It's a bit of a smell to need to do this kind of switch inline too.  I'd be tempted to do this:
```
type Scanner[CleanupV cleanupv1.ValueInterface] struct ...
```
Or if there are too many things to parameterise, uses a factory:
```
s.ipVersionedFactory.Value(...)
```